catalog/hyperledger/common.bom

```diff
@@ -1,70 +1,200 @@
-        [dockerrepo]
-        name=Docker Repository
-        baseurl=https://yum.dockerproject.org/repo/main/centos/$releasever/
-        enabled=1
-        gpgcheck=1
-        gpgkey=https://yum.dockerproject.org/gpg
-        EOF
-        sudo yum -y install docker-engine
-
-      post.install.command: |
-        # Configure Docker
-        sudo mkdir -p /etc/systemd/system/docker.service.d
-        echo "[Service]" | sudo tee --append /etc/systemd/system/docker.service.d/docker.conf > /dev/null
-        echo "ExecStart=" | sudo tee --append /etc/systemd/system/docker.service.d/docker.conf > /dev/null
-        echo 'ExecStart=/usr/bin/docker daemon -D --api-cors-header="*" --storage-driver=devicemapper -H tcp://0.0.0.0:4243 -H unix:///var/run/docker.sock' | sudo tee --append /etc/systemd/system/docker.service.d/docker.conf > /dev/null
-
-        sudo systemctl enable docker.service
-        sudo systemctl daemon-reload
-
-      launch.command: |
-        sudo service docker start
-
-      stop.command: |
-        sudo service docker stop
-
-      checkRunning.command: |
-        sudo service docker status
-
-      provisioning.properties:
-        osFamily: centos
-        minRam: 4gb
-        installDevUrandom: true
-        required.ports:
-          - 22
-          - 4243
-          - 7050
-          - 7051
-          - 7052
-          - 7053
-          - 7054
-          - 7055
-          - 7056
-          - 7057
-          - 7058
-          - 7059
-          - 8080
-          - 9999
-
-      childStartMode: foreground_late
+    - id: hyperledger-docker-engine
+      description: |
+        The Docker engine for running containers.
+      itemType: entity
+      item:
+        name: "hyperledger-docker-engine"
+        type: org.apache.brooklyn.entity.software.base.VanillaSoftwareProcess
+
+        brooklyn.parameters:
+          - name: docker.package
+            description: "Docker Package Name"
+            type: string
+            default: "docker-ce"
+          - name: docker.port
+            description: "Docker Package Number"
+            type: integer
+            default: 2376
+
+        brooklyn.config:
+          shell.env:
+            DOCKER_PACKAGE: $brooklyn:config("docker.package")
+            HOST_ADDRESS: $brooklyn:attributeWhenReady("host.address")
+            DOCKER_PORT: $brooklyn:config("docker.port")
+
+          install.command: |
+            sudo yum -y update
+            sudo yum install -y yum-utils device-mapper-persistent-data lvm2
+            sudo yum-config-manager --add-repo \
+              https://download.docker.com/linux/centos/docker-ce.repo
+            sudo yum-config-manager --enable extras
+            sudo yum-config-manager --enable epel-testing
+            sudo yum install -y ${DOCKER_PACKAGE}
+
+          post.install.command: |
+            echo "[HLF] docker installed"
+
+          customize.command: |
+            sudo tee /etc/docker/daemon.json <<EOF
```

Comment:
The file `/etc/docker` appears not to exist at this point
```
tee: /etc/docker/daemon.json: No such file or directory
```
Adding `sudo mkdir -p /etc/docker` before seems to fix this.